app/components/header-nav.js

```diff
@@ -1,70 +1,80 @@
 import { inject as service } from '@ember/service';
-import Component from '@ember/component';
+import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
-import { computed } from '@ember/object';
+import { action } from '@ember/object';
 
-export default Component.extend({
-  tagName: 'nav',
-  classNames: 'header-nav',
-  session: service(),
-  layoutManager: service('layout-manager'),
-  media: service(),
-  intl: service(),
-  localStorage: service(),
-  store: service(),
-  router: service(),
-  abilities: service(),
-  staticPages: tracked(),
+export default class HeaderNav extends Component {
+  tagName = 'nav';
```

Comment:
These are legacy Ember component API and Glimmer won't do anything with them. You need to wrap the stuff in the template in these accordingly: https://guides.emberjs.com/release/upgrading/current-edition/glimmer-components/#toc_outer-html